src/GovUk.Education.ExploreEducationStatistics.Common/Database/SqlServerDbContextOptionsBuilderExtensions.cs

```diff
@@ -0,0 +1,20 @@
+#nullable enable
+using System;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Database;
+
+public static class SqlServerDbContextOptionsBuilderExtensions
+{
+    public static SqlServerDbContextOptionsBuilder EnableCustomRetryOnFailure(
+        this SqlServerDbContextOptionsBuilder builder,
+        int? maxRetryCount = null,
+        TimeSpan? maxRetryDelay = null)
+    {
+        return builder.ExecutionStrategy(c =>
+            new MessageAwareSqlServerRetryingExecutionStrategy(
+                c, 
```

Comment:
I think `dependencies` might be a better variable name here, or whatever this actually is. I'm not sure what `c` is in this instance just from reading it.

Reply:
I couldn't agree more!  Updated.